### PR TITLE
[Interpreter Backend] Add support for BatchPermutation operator

### PR DESCRIPTION
**Pull request by rgopinath8**
Summary:
Adding support for BatchPermutation operator Interpreter backend implementation

Documentation: None

Test Plan: Ninja test is run, added  InterpreterOperatorTest cases

### CODE REVIEW

**Comment by rgopinath8** [comment]
Hi @jfix71 Kindly help to review it.

File: lib/Backends/Interpreter/InterpreterNodes.cpp [hunk 1]
```diff
@@ -6829,3 +6829,74 @@ void BoundInterpreterFunction::fwdExternalFunctionCallInst(
...
+  const dim_t elementSize = inputT->getType().getElementSize();
+  const dim_t dataSliceBytes = K * elementSize;
+
+  // For each batch input, calculate the corresponding index in the output
+  // based on indices and copy the data to output tensor.
+  // DistributeFPNPropasals op currently assigns index value as -1 for invalid
+  // ROIs and index value -1 is skipped, since the corresponding index in the
+  // output is filled with zeroes.
+  for (dim_t n = 0; n < batchSize; n++) {
+    if (indicesH.at({n}) != -1) {
```
Comment (jfix71) [inline lib/Backends/Interpreter/InterpreterNodes.cpp:6857]: nit: early continue to reduce indentation depth
```suggestion
    if (indicesH.at({n}) == -1) {
      continue;
    }
```

File: lib/Graph/Graph.cpp [hunk 2]
```diff
@@ -5185,6 +5185,22 @@ BBoxTransformNode *Function::createBBoxTransform(
...
+BatchPermutationNode *Function::createBatchPermutation(llvm::StringRef name,
+                                                       NodeValue input,
+                                                       NodeValue indices) {
+  auto inputDims = input.dims();
+  auto indicesDims = indices.dims();
+  std::vector<dim_t> outDims;
+  // Calculate output tensor shape from input and indices dimensions
+  for (auto dim : inputDims) {
+    outDims.push_back(dim);
+  }
```
Comment (jfix71) [inline lib/Graph/Graph.cpp:5197]:
```suggestion
  std::vector<dim_t> outDims(inputDims.begin(), inputDims.end());
  // Calculate output tensor shape from input and indices dimensions
```

**Review by jfix71 — APPROVED** [review]
LGTM! Added a couple small nits.

**Pull request closed without merging**